src/resolvers/Company.js

```diff
@@ -0,0 +1,89 @@
+import { ApolloError, UserInputError } from "apollo-server-express";
+import Company from "../models/Company";
+import User from "../models/User";
+
+const CompanyResolver = {
+  Query: {
+    companies: async(root, { filter } , context, info) => {
+      if (!filter){
+        throw new UserInputError("No filter provided");
+      }
+      try{
+        const filteredCompanies = await Company.find(JSON.parse(filter));
+        return filteredCompanies;

```

Comment:
Separate parsing and query:
- what if the parsed filter is invalid?
- surround parsing in its own try catch block